tests/query/processors/test_filter_in_select_optimizer.py

```diff
@@ -50,107 +59,241 @@ def subscriptable_reference(name: str, key: str) -> SubscriptableReference:
     )
 
 
-mql_test_cases: list[tuple[str, dict]] = [
-    (
-        "sum(`d:transactions/duration@millisecond`){status_code:200} / sum(`d:transactions/duration@millisecond`)",
-        {
-            Column("_snuba_metric_id", None, "metric_id"): {
-                Literal(None, 123456),
-            }
-        },
-    ),
-    (
-        "sum(`d:transactions/duration@millisecond`){status_code:200} / sum(`d:transactions/duration@second`)",
-        {
-            Column("_snuba_metric_id", None, "metric_id"): {
+def equals(lhs: Expression, rhs: Expression) -> FunctionCall:
+    return binary_condition("equals", lhs, rhs)
+
+
+def _and(lhs: FunctionCall, rhs: FunctionCall) -> FunctionCall:
+    return binary_condition("and", lhs, rhs)
+
+
+def _or(lhs: FunctionCall, rhs: FunctionCall) -> FunctionCall:
+    return binary_condition("or", lhs, rhs)
+
+
+def _in(lhs: Expression, rhs: Expression) -> FunctionCall:
+    return binary_condition("in", lhs, rhs)
+
+
+expected_optimize_condition = {
+    "sum(`d:transactions/duration@millisecond`){status_code:200} / sum(`d:transactions/duration@millisecond`)": _or(
```

Comment:
my unit tests verify: given an mql query, does the optimizer generate the expected condition